build-common.properties

```diff
@@ -72,6 +73,7 @@ checker.javadoc.lib=${checker.loc}/dist/checker-javadoc.jar
 
 checker.jdk8orhigher.sources=org/checkerframework/checker/nullness/Opt.java
 
+javaparser.core.lib=${checker.loc}/lib/javaparser-core.jar
```

Comment:
`checker/lib` contains annotated versions of libraries. This is a dependency, not an annotated library, so a different location should be used.
Also, the `framework` project depends on the javaparser, so this library shouldn't be in the `checker` project.